service/storage_proxy.hh

```diff
@@ -101,6 +101,37 @@ using allow_hints = bool_class<allow_hints_tag>;
 
 using is_cancellable = bool_class<struct cancellable_tag>;
 
+using storage_proxy_clock_type = lowres_clock;
+
+class storage_proxy_coordinator_query_options {
+    storage_proxy_clock_type::time_point _timeout;
+
+public:
+    service_permit permit;
+    client_state& cstate;
+    tracing::trace_state_ptr trace_state = nullptr;
+    replicas_per_token_range preferred_replicas;
+    std::optional<db::read_repair_decision> read_repair_decision;
+
+    storage_proxy_coordinator_query_options(storage_proxy_clock_type::time_point timeout,
+            service_permit permit_,
+            client_state& client_state_,
+            tracing::trace_state_ptr trace_state = nullptr,
+            replicas_per_token_range preferred_replicas = { },
+            std::optional<db::read_repair_decision> read_repair_decision = { })
+        : _timeout(timeout)
+        , permit(std::move(permit_))
+        , cstate(client_state_)
+        , trace_state(std::move(trace_state))
+        , preferred_replicas(std::move(preferred_replicas))
+        , read_repair_decision(read_repair_decision) {
+    }
+
+    storage_proxy_clock_type::time_point timeout(storage_proxy& sp) const {
+        return _timeout;
+    }
+};
+
```

Comment:
Not this patch's fault, but these options-style things are better as structs, not classes. But the series is right not to touch it.